src/app/InteractionModelEngine.cpp

```diff
@@ -1578,5 +1582,51 @@ void InteractionModelEngine::OnFabricRemoved(const FabricTable & fabricTable, Fa
     // the fabric removal, though, so they will fail when they try to actually send their command response
     // and will close at that point.
 }
+
+CHIP_ERROR InteractionModelEngine::ResumeSubscriptions()
+{
+#if CHIP_CONFIG_PERSIST_SUBSCRIPTIONS
+    SubscriptionResumptionStorage::SubscriptionIndex subscriberIndex;
+    CHIP_ERROR err = mpSubscriptionResumptionStorage->LoadIndex(subscriberIndex);
+    ChipLogProgress(InteractionModel, "%u subscriber nodes to resume.. (error %" CHIP_ERROR_FORMAT ")",
+                    static_cast<unsigned>(subscriberIndex.mSize), err.Format());
+    ReturnErrorOnFailure(err);
+
+    for (size_t i = 0; i < subscriberIndex.mSize; i++)
+    {
+        SubscriptionResumptionStorage::SubscriptionList subscriptions;
+
+        err = mpSubscriptionResumptionStorage->FindByScopedNodeId(subscriberIndex.mNodes[i], subscriptions);
+
+        ChipLogProgress(
+            InteractionModel, "\tNode " ChipLogFormatScopedNodeId ": Loaded %u subscriptions.. (error %" CHIP_ERROR_FORMAT ")",
+            ChipLogValueScopedNodeId(subscriberIndex.mNodes[i]), static_cast<unsigned>(subscriptions.mSize), err.Format());
+
+        ReturnErrorOnFailure(err);
+
+        for (size_t j = 0; j < subscriptions.mSize; j++)
+        {
+            SubscriptionResumptionStorage::SubscriptionInfo & subscriptionInfo = subscriptions.mSubscriptions[j];
+            auto requestedAttributePathCount                                   = subscriptionInfo.mAttributePaths.AllocatedCount();
+            auto requestedEventPathCount                                       = subscriptionInfo.mEventPaths.AllocatedCount();
+            if (!EnsureResourceForSubscription(subscriptionInfo.mFabricIndex, requestedAttributePathCount, requestedEventPathCount))
+            {
+                ChipLogProgress(InteractionModel, "no resource for Subscription resumption");
+                return CHIP_ERROR_NO_MEMORY;
```

Comment:
I'll add logic to delete subscription on error before return.

Reply:
Overall, the error could be set as:

```
CHIP_ERROR finalStatus = CHIP_NO_ERROR;

...

finalStatus = (finalStatus == CHIP_NO_ERROR) ? CHIP_ERROR_NO_MEMORY : finalStatus
```

That would allow "sticky first error", but then allow you to `continue` the loops, cleaning-up as you go, and then having a single exit at the bottom

Reply:
Given `EnsureResourceForSubscription` is unlikely to succeed in another call immediately after failure, I've left the logic to exit after one failure.

Also, in my response to another comment by Andrei, I believe that the intent here is to keep subscriptions to retry on next reboot if possible. It makes sense to only remove them when there's reason to believe the subscription can't be resumed in the future.